monad-rpc/src/handlers/debug_replay.rs

```diff
@@ -0,0 +1,284 @@
+            let txn = triedb_env
+                .get_transaction(block_key, tx_loc.tx_index)
+                .await
+                .map_err(|e| {
+                    JsonRpcError::internal_error(format!("error getting transaction: {}", e))
+                })?
+                .ok_or_else(|| JsonRpcError::internal_error("no transaction data".into()))?;
+            let (seq_number, block_id) = block_key.seq_num_block_id();
+            let raw_payload = match eth_call(
+                chain_id,
+                txn.tx,
+                header.header,
+                txn.sender,
+                seq_number.0,
+                block_id.map(|id| id.0 .0),
+                eth_call_executor,
+                &state_overrides,
+                tracer,
+                false,
+            )
+            .await
+            {
+                CallResult::Success(monad_ethcall::SuccessCallResult { output_data, .. }) => {
+                    output_data
+                }
+                CallResult::Failure(error) => {
+                    return Err(JsonRpcError::eth_call_error(error.message, error.data))
+                }
+                CallResult::Revert(result) => result.trace,
+            };
+            let v: serde_cbor::Value = serde_cbor::from_slice(&raw_payload)
+                .map_err(|e| JsonRpcError::internal_error(format!("cbor decode error: {}", e)))?;
+            serde_json::value::to_raw_value(&v).map_err(|e| {
+                JsonRpcError::internal_error(format!("json serialization error: {}", e))
+            })
+        }
+        ExecutionMode::Bulk => {
+            let block_key = match params.hash_or_tag() {
+                HashOrTag::Hash(block_hash) => {
+                    if let Some(block_num) = triedb_env
+                        .get_block_number_by_hash(block_key, block_hash.0)
+                        .await
+                        .map_err(JsonRpcError::internal_error)?
+                    {
+                        triedb_env.get_block_key(SeqNum(block_num))
+                    } else {
+                        return Err(JsonRpcError::internal_error(format!(
+                            "block not found: {:?}",
+                            block_hash
+                        )));
+                    }
+                }
+                HashOrTag::Tag(_) => block_key,
+            };
+            let txns = triedb_env.get_transactions(block_key).await.map_err(|e| {
+                JsonRpcError::internal_error(format!("error getting transactions: {}", e))
+            })?;
+            let (seq_number, block_id) = block_key.seq_num_block_id();
+            let mut results: Vec<serde_cbor::Value> = Vec::with_capacity(txns.len());
+            for txn in txns {
+                let tx_hash = txn.tx.signature_hash();
+                // TODO(dhil): May skew the eth_call statistics tracker.
+                let raw_payload = match eth_call(
```

Comment:
Will eth_call consider the case that state changes in tx0 in the block will affect tx1?